# Article  
Insights into invasive species from whole-genome resequencing ([link here](https://onlinelibrary.wiley.com/doi/10.1111/mec.15999))  

# Authors  
* Henry L. North
* Angela McGaughran
* Chris D. Jiggins

# Journal and Year  
Molecular Ecology (2021)  

# Highlights  
## The five themes in invasion biology that could benefit from high-throughput seq
We discuss five broad themes in invasion biology research that have involved, or could benefit from, high-throughput sequencing. 

These are: 
* the role of preinvasion adaptation in enabling subsequent spread (Part 2);  
* tools to reconstruct invasion routes in space (Part 3);  
* demographic inference to reconstruct the timing of invasion events, which also sheds light on the role of population bottlenecks during invasion (Part 4); 
* in situ adaptation following introduction to novel bioregions (Part 5); 
* and the role of hybridization and introgression during invasion (Part 6), which brings together all of the preceding themes. 

These key themes reflect the focus of existing WGR articles in invasion biology and span the temporal range of the invasion sequence (Figure 1).  

<img src="https://user-images.githubusercontent.com/22843614/128868162-6dcc78c4-7168-4900-bb3f-367095a9498c.png" width="70%">

## 1. Measuring the contribution of preinvasion selection to invasion success  
### 1.1 Time-series data can distinguish between species destined for invasion and those that adapt in situ
One of the oldest postulations in the field of invasion biology is that some species are “predisposed” to invasiveness (Baker, 1965). In other words, some traits that facilitate spread in a new environment are **not adaptations** that arise following colonization (see Part 5), but instead **exist in native** populations prior to transport.

For example, Kreiner et al. (2019) used WGR data to trace the evolutionary history of glyphosate resistance, which facilitates agricultural invasion success in introduced Canadian populations of Amaranthus tuberculatus (common waterhemp). A combination of demographic modelling (see Parts 3 and 4) and scans for selective sweeps demonstrated that populations in Essex County were **introduced from native** midwestern populations in the United States, and carried glyphosate resistance alleles that were selected for in the native population. Therefore, adaptation had occurred **prior to the invasion event**.

### 1.2 Balancing selection maintains adaptive diversity that can facilitate invasion
Unlike the example above, pre-existing adaptations that increase invasive spread in the invaded range can sometimes experience a **different selective regime** in the **native range**. Lee and Gelembiuk (2008) argue that **fluctuating selection** pressures in the native range (e.g., through regular disturbance events) can maintain either **genetic variation** or **phenotypic plasticity**, which can then be acted on by **positive selection** in **novel environments**.  

This prediction has been tested in the copepod species complex Eurytemora affinis. Several native populations from high-salinity environments have independently invaded freshwater ecosystems in North America. Seasonal fluctuations in salinity in the native environment, combined with several life history traits of E. affinis including overlapping generations, create **favourable conditions** for **balancing selection**. In the invaded range, a genome-wide scan showed parallel signatures of directional selection at ion transport genes in replicate populations (Figure 2a). Furthermore, the loci identified as being under positive selection in the introduced range also showed signatures of long-term balancing selection in the native range (Figure 2b), consistent with the hypothesis that **fluctuating selection** had maintained **genetic variation** that could enable invasive spread in a novel environment. 

## 2. Reconstructing invasion routes in space  
### 2.1 A phylogeographic perspective on invasive species and invasive genes  
A useful application of population genetics is to quantify patterns of isolation-by-distance to **identify** genetically distinct **management units**, **infer** the **source** population(s) of invasive species, and **estimate** the **timing of introduction(s)**

(...) WGR data were used to show that the cosmopolitan crop pest Plutella xylostella (diamondback moth) most likely originated in South America (You et al., 2020).  

(...) WGR data were used to infer the source populations of specific alleles that facilitated the successful colonization of Canadian agricultural landscapes by A. tuberculatus.
### 2.2 Making the most of genomic data  
Whole-genome resequencing increases our ability to infer the geographic origin of invasive populations. For relatively simple invasion scenarios, this increase in precision could be insignificant compared to other sequencing technologies, such as RRS, which may record a sufficient amount of information per sample to reach the same conclusion as WGR data.

This is relevant because geographic coverage is more important than genomic coverage when reconstructing invasion routes. Therefore, unless WGR is simultaneously being used to address a different research question (e.g., quantifying admixture during invasion or inferring the timing of an invasion event), a **greater individual sample size** with **reduced genomic representation** may often be **superior** to a **smaller sample size of WGR samples**.

WGR will probably **add most value** to geographic inference where there is a **weak signature of population structure**, or in studies of admixed invasive populations aiming to infer the **geographic origin** of **specific loci**.

# New tools to take a look  
* [Locator](https://onlinelibrary.wiley.com/servlet/linkout?suffix=null&dbid=16&doi=10.1111%2Fmec.15999&key=10.7554%2FeLife.54507): Predicting geographic location from genetic variation with deep neural networks

# New data types to take a look  

# Questions  
