# Article  
Identifying and removing haplotypic duplication in primary genome assemblies ([link here](https://academic.oup.com/bioinformatics/article/36/9/2896/5714742))  

# Authors  
* Dengfeng Guan
* Shane A McCarthy
* Jonathan Wood
* Kerstin Howe
* Yadong Wang
* Richard Durbin

# Journal and Year  
Bioinformatics (2020)  

# Highlights  
## 1. Pipeline description  
1. Long-read sequencing data is mapped onto the input draft assembly using minimap2 and read depth is calculated at each base position in the assembly;  
2. The software then uses the read-depth histogram to select a cutoff to separate haploid from diploid coverage depths, allowing for scenarios where the total assembly is dominated by either haploid or diploid sequence;  
    * for a haploid assembly, a bimodal distribution should be observed if duplication has occurred. The 0.5x read-depth peak results from the regions which are duplicated in the assembly (therefore half reads maps to one copy and the other half to another) and the 1x read-depth peak results from regions that are properly haplotype-fused;  
    * for a diploid assembly, as the entire assembly should be duplicated, the 1x peak may only be very small or not visible at all;  
3. The input draft assembly is split into contigs by cutting at blocks of 'N's. Minimap2 is then used to generate an all by all self-alignment;  
4. The haplotigs are then recognized and removed in essentially the same way as in *purge_haplotigs*, and all matches associated with haplotigs from self-alignment set are removed;  
    * if a high proportion of bases (>= 80%) are in the 1x read-depth range, the contig is labelled as a primary contig;  
    * otherwise, the contig is is considered a possible haplotig  
    * then, for each possible haplotig, we consider its best alignment to another contig  
    * if the alignment score is greater than a certain threshold (default: 70%), the contig is marked as a haplotig  
        * however, if the alignment score is greater than 70% **and** the max match score largent than 200%, it is marked as a repeat to highlight potential problematic contigs such as collapsed repeats or low-complexity regions  
     
5. Finally  

## 2. Quality analysis  
Two analysis were done in order to compare *purge_dups* with other two tools traditionally used for haplotigs removal: *purge_haplotigs* and *HaploMerger2*:  
1. K-mer comparison analysis using the software KAT  
    * goal: to assess the efficiency of removal of false duplications in the assembly
2. Benchmarking Universal Single-Copy Orthologs (BUSCO)  
    * goal: to assess the consequences of purging for gene set completeness and duplication 

## 3. Comparison with other purging tools  
* *purge_dups* removes 96.4% of duplicated haploid-unique k-mers (...), comparable figures for *HaploMerger2* and *purge_haplotigs* are 95.7% and 81.2%, respectively;  
* Regarding the BUSCO analyses, all purging methods removed a substancial amount of sequence from the primary assembly and decreased BUSCO duplication, and no single method performed uniformly betterr across all assemblies and metrics tested. 
    * However ***purge_haplotigs*** consistently leaves *more duplicated sequence and genes*. 
    * For all assemblies other than Mm (*Myripristis murdjan*), ***purge_dups*** gives the **highest** fraction of **single-copy complete** genes, and the **lowest** fraction of **missing genes**.  


# New tools to take a look  
* *Genomescope*: fast reference-free genome profiling from short reads. ([Crossref](http://dx.doi.org/10.1093/bioinformatics/btx153))  

# Questions  
* 


