# Article  
Identifying and removing haplotypic duplication in primary genome assemblies  

# Authors  
* Dengfeng Guan
* Shane A McCarthy
* Jonathan Wood
* Kerstin Howe
* Yadong Wang
* Richard Durbin

# Journal and Year  
Bioinformatics (2020)  

# Highlights  
## 1. Pipeline description  
1. Long-read sequencing data is mapped onto the input draft assembly using minimap2 and read depth is calculated at each base position in the assembly;  
2. The software then uses the read-depth histogram to select a cutoff to separate haploid from diploid coverage depths, allowing for scenarios where the total assembly is dominated by either haploid or diploid sequence;  
3. The input draft assembly is split into contigs by cutting at blocks of 'N's. Minimap2 is then used to generate an all by all self-alignment;  
4. The haplotigs are then recognized and removed in essentially the same way as in *purge_haplotigs*, and all matches associated with haplotigs from self-alignment set are removed;  
5. Finally  

## Quality analysis  
Two analysis were done in order to compare *purge_dups* with other two tools traditionally used for haplotigs removal: *purge_haplotigs* and *HaploMerger2*:  
1. K-mer comparison analysis using the software KAT  
  * goal: to assess the efficiency of removal of false duplications in the assembly
2. Benchmarking Universal Single-Copy Orthologs (BUSCO)  
  * goal: to assess the consequences of purging for gene set completeness and duplication 

# New tools to take a look  

# New data types to take a look  

# Questions  
* 


