# Article  
Inferring synteny between genome assemblies: a systematic evaluation ([link here](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2026-4))  

# Authors  
* Dang Liu  
* Martin Hunt  
* Isheng J Tsai  

# Journal and Year  
BMC Bioinformatics (2018)  

# Highlights  
## Limitations of synteny algorithms to deal with fragmented assemblies  
* "Synteny analysis is regularly performed on assembled sequences that are fragmented, neglecting the fact that most methods were developed using complete genomes. It is unknown to what extent draft assemblies lead to errors in such analysis."  
* "Analysis of synteny in closely related species is now the norm for every new published genome. However, assembly quality comes into question as it has been demonstrated to affect subsequent analysis such as annotation or rate of lateral transfer [32, 33]."
* "We fragmented genome assemblies of model nematodes to various extents and conducted synteny identification and downstream analysis. We first show that synteny between species can be underestimated up to 40% and find disagreements between popular tools that infer synteny blocks."  
* "Our results show that a minimum N50 of 1 Mb is required for robust downstream synteny analysis, which emphasizes the importance of gold standard genomes to the science community, and should be achieved given the current progress in sequencing technology."  

## Synteny blocks  
* "Synteny blocks are more formally defined as regions of chromosomes between genomes that share a common order of homologous genes derived from a common ancestor"  
* " Alternative names such as **conserved synteny** or **collinearity** have been used interchangeably"  

## Why not just align the whole genomes?  
* "In general, synteny identification is a filtering and organizing process of all local similarities between genome sequences into a coherent global picture [34]."  
* "The most intuitive way to identify synteny would be to establish from selective genome alignments [35, 36], but levels of nucleotide divergence between species may make such methodologies challenging."  
* "Instead, many tools use orthologous relationships between protein-coding genes as anchors to position statistically significant local alignments."

# New tools to take a look  

# New data types to take a look  

# Questions  
