# Article  
HiCanu: accurate assembly of segmental duplications, satellites, and allelic variants from high-fidelity long reads ([link here](https://doi.org/10.1101/gr.263566.120))  

# Authors  
* Sergey Nurk
* Brian P. Walenz
* Arang Rhie
* Mitchell R. Vollger
* Glennis A. Logsdon
* Robert Grothe
* Karen H. Miga
* Evan E. Eichler
* Adam M. Phillippy
* Sergey Koren

# Journal and Year  
Genome Research (2020)  

# Highlights  
## Advantages of HiFi data  
* Although HiFi reads are shorter (up to 25 Kb) than the traditional CLR reads (that can exceed 50 Kb) or the Nanopore reads (that can exceed 100 Kb), HiFi reads are much more accurate compared to other long-read technologies. While the median accuracy of the former can exceed 99.9% (>Q30), the accuracy of the latter do not exceed 95%. 
## HiCanu vs. Canu  
* HiCanu was built on the original Canu pipeline, which was developed for dealing with conventional long-read data, but it is not optimized to deal with HiFi data. HiCanu was built after replacement and significant modification of Canu's core methods in order to better deal with HiFi data. Here are some key steps in the HiCanu pipeline:  
    1. HiCanu begins by compressing consecutive copies of the same nucleotide (homopolymers) to a single base (e.g. "AA..." becomes "A").  
        * This is done because homopolymer length is the primary error mode of HiFi technology  
    2. The compressed reads are trimmed based on their overlaps to other reads to remove any chimeric sequences or sequencing adapters, and the overlaps are recomputed based on the trimmed reads
    3. The Overlap Error Adjustment (OEA) module examines read overlap pileups to identify remaining sequencing errors in the individual reads and recomputes overlap alignment identities. Here, a modification was applied to the OEA module to ignore any differences within microsatellite repeat arrays due to the fact that they are also prone to HiFi read errors. Then two reads that differ only by an indel detected in a microsatellite region are considered to be 100% identical.   
    <img src="https://user-images.githubusercontent.com/22843614/92035046-826d9480-ed44-11ea-8427-316d65319521.png" width=30%></img>  
    *Figure 1. A) Two hypothetical reads are shown with sequencing errors highlighted in red. B) The first step of HiCanu is to compress homopolymers, which obscures homopolymer length errors but retains enough information to accurately distinguish reads from different genomic loci. C) Overlaps are then computed for the compressed reads, and remaining errors are identified by examining the alignment pileups (gray rectangle). D) Finally, after correcting the identified errors (blue) and ignoring indels in regions of known systematic error (gray), the resulting overlap is 100% identical.*  
    4. The adjusted overlaps are then used to built draft contigs using Canu's Bogart module, modified to better handle heterozygous variants and consider only high-identity read overlaps  
    5. Final contigs are obttained by computing a consensus across the original, uncompressed reads, arranged according to the layout of their compressed versions. 

* Similar to many modern assemblers, when faced with a diploid genome, HiCanu outputs contigs as "pseudo-haplotypes" that preserve local allelic phasing but may switch between haplotypes across longer distances.
    * A single set of contigs representing all resolved alleles is output regardless of ploidy, and **additional processing** with a tool such as **Purge_dups** is required to partition the contigs into primary and alternate allele sets.  

## *Drosophila* genome assembly  
* Assembly was tested in four conditions:  
    1. 40x coverage HiFi assembled with HiCanu  
    2. 40x coverage HiFi assembled with Peregrine
    3. 40x coverage HiFi assembled with Canu  
    4. 200x coverage CLR assembled with Canu  
    
* After assembly, contigs **less than 50 kbp** were **filtered** in order to exclude **low-quality** sequences consisting of only a **few reads**.  

* With exception of the Peregrine assembly, all other assemblies supported the hypothesis that alleles from both haplotypes were being represented in the final assembly. This was indicated by:  
    * The assembly length being more than twice that of the *Drosophila* haploid reference genome  
    * The large fraction of duplicated genes observed on BUSCO analysis  

* To allow like-for-like comparisons, all assemblies were processed with Purge_dups to separate primary and alternate contig sets. Assembly statistics were then calculated for both contig sets separately. The only analysis where the assembly used was the diploid one (i.e. before purge_dups) was the structural correctness, since purge_dups may introduce (or correct) mis-assemblies as it modifies the contigs.   

### On the meaning of "Pseudo-haplotypes"  
* HiFi reads alone cannot be used to infer phasing across homozygous regions longer than the read length, so the contigs produced by HiCanu (and Canu) represent "pseudo-haplotypes", which may switch between haplotypes.  
* However, for highly heterozygous genomes with short regions of homozygosity, HiCanu is expected to produce a low number of haplotype switches and mostly preserve long-range phasing.  

## Human genome assemblies  
* Three different libraries where tested, all consisting of approximetely 30x HiFi sequencing coverage:  
    1. 20 kbp library of the completely homozygous cell line CHM13  
    2. 15 kbp library of the Ashkenazic cell line HG002 (from the Personal Genome Project)  
    3. combined library (10 kbp, 15 kbp and 20 kbp fragments sizes) of the Puerto Rican cell line HG00733 (from the 1000 Genomes Project)     
* Then, for each library, three different assemblers were tested: HiCanu, Canu and Peregrine
* For each cell line tested, the most continuous published assembly was also considered in the comparative analysis of assembly strategies. Those assemblies relied on ultra-long Oxford Nanopore (ONT) reads to achieve state-of-the-art repeat resolution  
* As in the Drosophila evaluation, contigs less than 50 kbp were excluded from analysis and the diploid assemblies generated by HiCanu and Canu were broken into primary and alternate contig sets, that were evaluated separately.  

### Continuity  
* The continuity of HiCanu assemblies, as measured by NG50, exceeded that of all other HiFi-based assemblies and even rivaled the continuity of Nanopore ultra-long read assemblies  

### Consensus accuracy  
* For consensus accuracy, the HiCanu primary contig sets exceeded QV50 (99.999% accuracy) and alternate contigs sets exceeded QV40 (99.99% accuracy), while the unpolished Nanopore assemblies failed to exceed QV30 (99.9%).  

## To polish or not to polish?  
* Although Nanopore assemblies currently require polishing with complementary technologies to maximize consensus accuracy, we discourage polishing HiCanu HiFi assemblies, because the available polishing pipelines may map reads back to the wrong repeat copies and actually introduce errors during polishing.

